### PR TITLE
320 rounds: fix restoring from a catchpoint

### PR DESCRIPTION
**Pull request by tolikzinovyev**
## Summary

Restore catchpoints into a V6 database and let the migration update the database.

## Test Plan

Manually ran fast catchup.

### CODE REVIEW

File: ledger/accountdb.go [hunk 1]
```diff
@@ -1130,26 +1124,15 @@ func resetCatchpointStagingBalances(ctx context.Context, tx *sql.Tx, newCatchup
 // tables and update the correct balance round. This is the final step in switching onto the new catchpoint round.
 func applyCatchpointStagingBalances(ctx context.Context, tx *sql.Tx, balancesRound basics.Round, merkleRootRound basics.Round) (err error) {
 	stmts := []string{
-		"ALTER TABLE accountbase RENAME TO accountbase_old",
-		"ALTER TABLE assetcreators RENAME TO assetcreators_old",
-		"ALTER TABLE accounthashes RENAME TO accounthashes_old",
-		"ALTER TABLE resources RENAME TO resources_old",
-		"ALTER TABLE onlineaccounts RENAME TO onlineaccounts_old",
-		"ALTER TABLE txtail RENAME TO txtail_old",
+		"DROP TABLE IF EXISTS accountbase",
```
Comment (pzbitskiy) [inline ledger/accountdb.go:1127]: did you test it and tables survive after rolling back the transaction?

File: ledger/catchupaccessor.go [hunk 2]
```diff
@@ -821,6 +821,26 @@ func (c *CatchpointCatchupAccessorImpl) finishBalances(ctx context.Context) (err
 			}
 		}
 
+		// Reset the database to version 6.
+		// TODO: pass ctx.
+		err = accountsReset(tx)
+		if err != nil {
+			return err
+		}
+		{
```
Comment (pzbitskiy) [inline ledger/catchupaccessor.go:830]: scoping does not buy anything here

File: ledger/accountdb.go (hunk 1 above)
Comment (tolikzinovyev) [reply ledger/accountdb.go:1127]: just tested it

File: ledger/catchupaccessor.go (hunk 2 above)
Comment (tolikzinovyev) [reply ledger/catchupaccessor.go:834]: it hides `tp`

File: ledger/catchupaccessor.go [hunk 3]
```diff
@@ -821,6 +821,26 @@ func (c *CatchpointCatchupAccessorImpl) finishBalances(ctx context.Context) (err
...
+			tp := trackerDBParams{
+				initAccounts:      c.ledger.GenesisAccounts(),
+				initProto:         c.ledger.GenesisProtoVersion(),
+				catchpointEnabled: c.ledger.catchpoint.catchpointEnabled(),
+				dbPathPrefix:      c.ledger.catchpoint.dbDirectory,
+				blockDb:           c.ledger.blockDBs,
+			}
+			_, err = runMigrations(ctx, tx, tp, c.ledger.log, 6)
```
[the remark this answers is not in the document]
Comment (tolikzinovyev) [reply ledger/catchupaccessor.go]: done

File: ledger/catchupaccessor.go (hunk 2 above)
Comment (algorandskiy) [reply ledger/catchupaccessor.go:834]: please remove, the function is trivial

File: ledger/catchupaccessor.go [hunk 4]
```diff
@@ -821,6 +821,30 @@ func (c *CatchpointCatchupAccessorImpl) finishBalances(ctx context.Context) (err
...
+		}
+		{
+			tp := trackerDBParams{
+				initAccounts:      c.ledger.GenesisAccounts(),
+				initProto:         c.ledger.GenesisProtoVersion(),
+				catchpointEnabled: c.ledger.catchpoint.catchpointEnabled(),
+				dbPathPrefix:      c.ledger.catchpoint.dbDirectory,
+				blockDb:           c.ledger.blockDBs,
+			}
+			_, err = runMigrations(ctx, tx, tp, c.ledger.log, 6 /*target database version*/)
```
Comment (algorandskiy) [inline ledger/catchupaccessor.go:842]: `const targetDBVersion = 6` + comment about what happens here, and how to support version 8

**Review by algorandskiy — CHANGES_REQUESTED** [review]

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**